### ISSUE TITLE
Program synthesis data, reproduce in t5x and data proportion

### ISSUE THREAD

**Issue by nawnoes** [google-research/FLAN#26]
Thanks for a nice open source and paper.

I would like to reproduce the FLAN-T5. I have a few questions.

1. I can't see "Program Synthesis" in the flan v2 data, how can I add it?  

2. I built the mixture following this repo within t5x. It takes about 2 days to train flan-t5 in t5x. 
    I wonder it takes a long time to build the mixture. should I make a cache for FLAN-T5 training?

3. what is the proportion of data used in the published FLAN-T5?
  - Mixed zero, few-shot prompt proportion - zero-shot: few-shot = 10:90 or 50:50
  - Submixture proportion: flan(46%), t0(28%), cot(1.8%), niv2(24.2%), dialog(...), program synthesis(...)

Thanks!

**Comment by shayne-longpre** [comment]
@nawnoes thanks for the questions! 

1. Unfortunately due to legal reasons we had to omit a couple datasets from the generation scripts when porting from the internal version. This includes the Dr Repair datasets (https://github.com/michiyasunaga/DrRepair) and Deepmind Code Contests (https://github.com/deepmind/code_contests), and Task Master (https://github.com/google-research-datasets/Taskmaster) from the Dialog submixture. You're welcome to add them for your own purposes, but overall these are a small fraction of training, and so should only have negligible effect if omitted, unless of course your target eval is program synthesis. 

2. Yes, I would definitely recommend caching it! I'll note that we found a bug, identified in #25, with the few-shot splits of our NIv2 split. This bug was introduced after we ported it to TFDS, so it wasn't in our internal version. We are working on a fix.

3. We actually used (25/25/25/25) across (zsopt/zsnoopt/fsopt/fsnoopt) for each split. Without Program Synthesis I would recommend roughly: 40% Flan, 32% T0, 20% NIv2, 5% CoT, 3% Dialog.

I will add these details on the README soon! Thanks again,
